\documentclass[oneside]{book}
\usepackage{tests-tex-notebook}

\begin{document}

\chapter{Case with explicit error}

\begin{pycell}
raise RuntimeError("this is an error")
\end{pycell}

Define a variable that will be cleared out before we move to the next chapter

\begin{pycell}
a = 1
\end{pycell}

\begin{pycell}
print(a)
\end{pycell}

This correctly prints the value of variable.

\chapter{Variable resets}

In this file the python session resets at every chapter, and therefore

\begin{pycell}
a
\end{pycell}

fails with a \texttt{NameError}.

\ifPythonTeXLoaded
% nothing else to be done
\else
% Mock failure in python execution, since pythontex is not running
\printfailfilelinebyline{expected/\jobname.err.txt}
\fi

\end{document}
